### PR TITLE
Switchboard

### PR DESCRIPTION
**Pull request by eamsden**
This PR implements the core functionality for urbit-webrtc: call signalling over Urbit.

The driver is the `switchboard` Gall agent, which is intended to be subscribed to either directly via airlock from a Javascript browser application, or from other native Urbit applications. Switchboard passes SDP messages in WebRTC's format between two instances running on two ships, allowing the subscribed applications to exchange SDP offers, answers, and ICE candidates.

The `switchboard-js` library implements these subscriptions in a factory class and an extension of the WebRTC-specified `RTCPeerConnection` class. Using the factory, web applications can easily both place calls to and receive calls from their counterparts attached to other ships, using only the name of the ship and the name of the application.

Once this connection is set up, it is only necessary to add media or data channels and connect them to the application UI.
A simple demonstration using data channels for chat can be found in the `urchatfm` React application.

### CODE REVIEW

**Comment by eamsden** [comment]
@tacryt-socryp I would appreciate another review before I merge this, if you have time. I'd like to merge it tomorrow evening (4/26 EDT)

**Comment by tacryt-socryp** [comment]
Will do tomorrow morning / early afternoon.

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:149]: indent 149

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:168]: indent 168

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:195]: this comment (and all the ones like it that are on the same indentation as the following case statement) should be dedented by two

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:204]: should be dedented by one

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:254]: indent

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:259]: indent 259-261 by one

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:274]: no new lines allowed

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:302]: Do you use your bowl? It doesn't look like it. If you don't, then you should just write this as a core `|%`.

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [inline switchboard/app/switchboard.hoon:428]: dedent 428-432, as `|=` has a fixed number of arguments (2)

**Review by tacryt-socryp — CHANGES_REQUESTED** [review]
Much better!

**Comment by tacryt-socryp** [comment]
Happy to call and discuss if anything is unclear as per JSON parsing (getting rid of `type` etc)

File: switchboard/app/switchboard.hoon
Comment (tacryt-socryp) [reply switchboard/app/switchboard.hoon:428]: Also use `%+` for turn instead of `%:`

File: switchboard/lib/switchboard.hoon
Comment (eamsden) [reply switchboard/lib/switchboard.hoon:49]: I will probably make this a separate PR, since removing the `type` variable will also involve matching changes to the JS library

File: switchboard/lib/switchboard.hoon
Comment (eamsden) [reply switchboard/lib/switchboard.hoon:68]: The result of `ot` is a gate with a `json` sample, but we cast it to just `call-signal:switchboard`? I fear I do not yet fully understand...

File: switchboard/lib/switchboard.hoon
Comment (eamsden) [reply switchboard/lib/switchboard.hoon:73]: See above re: separate PR

File: switchboard/lib/switchboard.hoon
Comment (tacryt-socryp) [reply switchboard/lib/switchboard.hoon:68]: Ah shit, no need to cast at all. You'll get the right result if you don't do the cast

**Pull request merged**
